packages/node_modules/@webex/plugin-meetings/src/peer-connection-manager/index.js

```diff
@@ -163,7 +163,10 @@ pc.iceCandidate = (peerConnection, {remoteQualityLevel}) =>
 
       if (!validateSdp(peerConnection.sdp)) {
         setTimeout(() => {
-          resolve(peerConnection);
+          // peerconnection does gather ice candidate IP but in some cases due to firewall 
+          // or proxy the ice candidate does not get gathered so we need to wait and then retry 
+          // if still not valid then throw an error saying missing ice candidate
+          reject(new MediaError('No ice candidate or port found in the SDP'));
```

Comment:
This is going to always reject after a timeout, we need to check again:
```suggestion
         if (validateSdp(peerConnection.sdp)) {
           resolve(peerConnection);
         } else {
           LoggerProxy.logger.error('PeerConnectionManager:index#iceCandidate --> SDP not valid after waiting.');
           reject(new MediaError('No ice candidate or port found in the SDP'));
         }
```

Reply:
you are correct